### ISSUE TITLE
Missing label on search form

### ISSUE THREAD

**Issue by dmundra** [readthedocs/sphinx_rtd_theme#1379]
### Problem

The search form doesn't have a required label. This is an accessibility issue, "a form control does not have a properly associated text label, the function or purpose of that form control may not be presented to screen reader users."

### Reproducible Project

You can see the error reported by the Wave tool: https://wave.webaim.org/report#/https://sphinx-rtd-theme.readthedocs.io/en/stable/.

#### Error Logs/Results

![Wave error report](https://user-images.githubusercontent.com/7096681/202043044-0169b108-2325-470a-be20-d995a88628a8.png)

#### Expected Results

Use `aria-label` to add the missing label. See https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html for examples. I will submit a pull request when I can.

**Comment by dmundra** [comment]
@stsewd I missed that. Thanks for linking it. It would be good to implement one of solutions so as to fix the issue.

**Comment by stsewd** [comment]
> @stsewd I missed that. Thanks for linking it. It would be good to implement one of solutions so as to fix the issue.

Your PR is a good to fix this problem soon, I think we should merge it. Mine is more a deeper change to match with upstream.
